x-pack/plugins/maps/public/routing/routes/list/load_list_and_render.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
+import { getToasts } from '../../../kibana_services';
+import { i18n } from '@kbn/i18n';
+import { MapsListView } from './list';
+import { Redirect } from 'react-router-dom';
+
+export const LoadListAndRender = class extends React.Component {
+  state = {
+    hasSavedMaps: null,
+    savedMapsList: null,
+  };
+
+  componentDidMount() {
+    this.loadMapsList();
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async loadMapsList() {
+    try {
+      const { hits = [] } = await getMapsSavedObjectLoader().find();
```

Comment:
By default getMapsSavedObjectLoader().find() uses a page size of 1000. This check could return a very large payload when you really are just checking that one exists. How about changing to `const { hits = [] } = await getMapsSavedObjectLoader().find('', 1);` to just pull a single saved object.